### DISCUSSION TITLE
HUD to show clip name and frame number instead of otio file refrence.

### DISCUSSION THREAD

**Discussion by cgene75** [ggarra13/mrv2#324] [Ideas]
I've noticed that when you playback an otio timeline, the HUD displays the refrence to the OTIO file, instead of each individual clip currently being played.

I think it would be super handy for the hud to reflect the info based on the location of the playhead instead:

![otio_timeline_hud_v01](https://github.com/user-attachments/assets/bf937762-7857-483f-b882-ac0986b2bf6b)

ps: it would also be super cool if you could see the 'image' menu there as well, for each clip the timeline to be able to compare the 'current' version to previous versions.

**Comment by ggarra13** [comment]
> I think it would be super handy for the hud to reflect the info based on the location of the playhead instead:

That should be relative easy to add.  Something like:

davinci_fcpmxl.otio (Soda_Pictures.0001.exr)

or:

davinci_fcpmxl.otio (Soda_Clip.mov)

> ps: it would also be super cool if you could see the 'image' menu there as well, for each clip the timeline to be able to compare the 'current' version to previous versions.

That, however, it is tricky.  I'll need to dig into the source code to try to figure it out if it is even possible, as the check would impact playback performance possibly dramatically.

**Comment by cgene75** [comment]
FWIW, I would fully expect a clip version change to invalidate the cache.

**Comment by cgene75** [comment]
I'm not sure if you prefer to to have a separate thread for each distinct idea, or whether you'd prefer them to be group into 'related thoughts' threads.  I'll continue here...

It would be super handy to be able to quickly set the in and out marks to a single shot's boundaries.

eg: you are playing back a sequence cam010, cam020, cam030, cam040 etc... and you wanna take a closer look at cam020... you pause the playhead on cam020, hit a hotkey, and the in an out marks set to the boundaries of cam020 in the timeline.  Hit the hotkey again, and the in and out marks clear.

**Comment by ggarra13** [comment]
Yes.  Actually, if instead of posting here in ideas, you report them as Issues (tagging them as a Feature), that would be great.  That way I can keep track of it, disscuss them and close the Issue once the feature is added.

**Comment by ggarra13** [comment]
Your three suggestions have been added to mrv2 v1.3.1 beta.  You can give them a go at:

https://sourceforge.net/projects/mrv2/files/beta/

**Comment by cgene75** [comment]
Heya Gonzalo,

Thanks for implementing these.  I'll be sure to log further RFE's as issues (tagging them as a feature).
The filename HUD is super useful.  Would it be possible to have the Frames, FrameRange, and FrameCount fields also reflect the 'child' data as well as the parent otio?

To round out this topic though, I do get an instacrash whenever I try any of the image->version menus.

**Reply by ggarra13** [reply]
> To round out this topic though, I do get an instacrash whenever I try any of the image->version menus.

Can you try the latest on sourceforge to see if you still get crashes.  Also, in case you do, let me know what platform.

**Reply by cgene75** [reply]
It's still crashing on windows11 on build: 
`mrv2 64 bits - v1.3.1 - Built Dec 20 2024 18:58:38`

I'm unable to test on linux yet, as I i'm still tweaking the xml translator, and don't have a file with viable paths.  i'll test on linux as soon as I can.

**Reply by ggarra13** [reply]
Please try this version that has Debug information.  When it crashes, it should save a $HOME/.filmaura/crash.log.

.exe:
https://mega.nz/file/qfwRmLpT#54P5aE5wja7kxKBeT2qe4AFw5kE1pLAwwAyB8SN4lJs

**Reply by ggarra13** [reply]
Also, give me the .otio file it is crashing with.

**Comment by cgene75** [comment]
Heya Gonzalo,

I've tested versioning up and down using build: 
mrv2 64 bits - v1.3.1 - Built Dec 20 2024 20:40:46

Unfortunately, I don't see a log file being created in $HOME/.filmaura
the only files in that folder are:

mrv2.keys.prefs
mrv2.ocio.json
mrv2.paths.prefs
mrv2.prefs

I'm attaching a single track otio file that i'm using to test with.

[davinci_fcpxml_single_track_v01.zip](https://github.com/user-attachments/files/18218951/davinci_fcpxml_single_track_v01.zip)

**Reply by ggarra13** [reply]
Sorry.  My bad.  The crash goes to %TMP%. On Windows, this is usually:  C:\Users\<USER>\AppData\Local\Temp.
Check there.

**Comment by cgene75** [comment]
see mrv2.crash.log attached:

[mrv2.crash.log](https://github.com/user-attachments/files/18219100/mrv2.crash.log)

**Comment by ggarra13** [comment]
This one should not crash.

.exe:
https://mega.nz/file/TXJxnZZQ#kB3BT-v8L6AiBLYCW-TyQVZpQyU9v-AqU4N7WXwm9J4

Note that if you are using mrv2 to create an .otio playlist, you can also do it command-line (and it will create an EDL*.otio file in $TMP).

`
mrv2 -edl <file1> <file2> ... etc...
`

**Comment by cgene75** [comment]
It's working!  Thank you for sorting this out!

It's great to know about being able to use the CLI flag to write otio files.  Very handy!  I'm currently creating otio files via an _in-progress_ fcpxml translator...
